### ISSUE TITLE
Compute forces on the IPU

### ISSUE THREAD

**Issue by AlexanderMath** [openmm/spice-dataset#80]
nanoDFT computes forces on the CPU using `def grad(..)` on [line 230](https://github.com/graphcore-research/pyscf-ipu/blob/3eaeaebf7bdf29cd592f87db2c2210c7d0025d68/nanoDFT/nanoDFT.py#L267). To run this on the IPU we could port [lines 269-273](https://github.com/graphcore-research/pyscf-ipu/blob/3eaeaebf7bdf29cd592f87db2c2210c7d0025d68/nanoDFT/nanoDFT.py#L269) and [line 283](https://github.com/graphcore-research/pyscf-ipu/blob/3eaeaebf7bdf29cd592f87db2c2210c7d0025d68/nanoDFT/nanoDFT.py#L283). 

Different strategies: 
1. Compile libcint to poplar and replace all `mol.intor(..)` with corresponding poplar calls (ERI is only real problem). 
2. Use Jax implementation from [D4FT](https://github.com/sail-sg/d4ft/tree/main/d4ft/integral) for the forward pass of the `mol.intor(..)` and match up the derivative with the calls on lines 270-273 ([pyscfad](https://github.com/fishjojo/pyscfad) did this on CPU so their code may be helpful).  
3. Line 230 uses [this theorem](https://faculty.washington.edu/miller/51816/Feynman.pdf) to compute gradients. We could instead use `jax.grad(..)`. For that we need to fix all calls in `_nanoDFT(..)` which don't support derivatives.

**Comment by AlexanderMath** [comment]
apologies, clicked post on the wrong github. if anyone has delete powers, please feel free to delete.
